### PR TITLE
Show pre submission reference guidance under not_requsted_yet status

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

At the moment, we have static guidance about contacting referees on this page: 

https://bat-design-history.netlify.app/images/apply-for-teacher-training/improving-the-references-user-journey/static-guidance-about-contacting-your-referees-when-you-submit.png

We've changed the design so that guidance appears under the referee status tag instead, appearing only in the 'not requested' state - i.e., when someone has added or replaced a reference:

https://bat-design-history.netlify.app/images/apply-for-teacher-training/improving-the-references-user-journey/showing-guidance-at-the-relevant-point-when-replacing-referees.png

## Changes proposed in this pull request

- Only show the pre-submission guidance when a reference is a  not_requsted_yet state

Before

image

After

image

Doesn't render post submission

image

## Guidance to review

I'll leave a comment about one thing 👍 

## Link to Trello card

https://trello.com/c/el
2X7sKpK/1442-dev-changes-to-references-section-for-apply-1-and-apply-2-turning-static-guidance-into-conditional-guidance-with-slight-change-t

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/components/candidate_interface/referees_review_component.rb [hunk 1]
```diff
@@ -69,16 +69,18 @@ def reference_type_row(referee)
...
-    def show_status?(referee)
-      referee.application_form.submitted?
+      if referee.not_requested_yet? && !referee.application_form.submitted?
+        {
+          key: 'Status',
+          value: feedback_status_label(referee) +
+            content_tag(:p, t('application_form.referees.info.not_requested_yet'), class: 'govuk-body govuk-!-margin-top-2'),
+        }
+      else
+        {
```
Comment (davidgisbey) [inline app/components/candidate_interface/referees_review_component.rb:79]: Is there any other situation where we wouldn't want to render the status? I can't think of one.

Comment (stevehook) [reply app/components/candidate_interface/referees_review_component.rb:79]: Can't think of one

File: spec/components/candidate_interface/referees_review_component_spec.rb [hunk 2]
```diff
@@ -30,9 +30,13 @@
...
     it 'renders component with correct value for status for unrequested reference' do
       application_form.update_column(:submitted_at, nil)
+      first_referee = application_form.application_references.first
+      first_referee.update(feedback_status: 'not_requested_yet')
       result = render_inline(described_class.new(application_form: application_form))
 
-      expect(result.css('.govuk-summary-list__key').text).not_to include('Status')
+      expect(result.css('.govuk-summary-list__key').text).to include('Status')
+      expect(result.css('.govuk-summary-list__value').to_html).to include('Not requested')
+      expect(result.css('.govuk-summary-list__value').to_html).to include(t('application_form.referees.info.not_requested_yet'))
```
Comment (stevehook) [inline spec/components/candidate_interface/referees_review_component_spec.rb:39]: Would be good to also see a couple of specs asserting this copy is not present when the status is different or the application is submitted.

Comment (davidgisbey) [reply spec/components/candidate_interface/referees_review_component_spec.rb:39]: Good shout. I've added a couple in.

**Pull request merged**